extensions/objects/templates/ListOfSetsOfHtmlStringsEditor.js

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+oppia.directive('listOfSetsOfHtmlStringsEditor', [
+  '$compile', 'OBJECT_EDITOR_URL_PREFIX',
+  function($compile, OBJECT_EDITOR_URL_PREFIX) {
+    return {
+      link: function(scope, element) {
+        scope.getTemplateUrl = function() {
+          return OBJECT_EDITOR_URL_PREFIX + 'ListOfSetsOfHtmlStrings';
+        };
+        $compile(element.contents())(scope);
+      },
+      restrict: 'E',
+      scope: {
+        getInitArgs: '&',
+        value: '='
+      },
+      template: '<span ng-include="getTemplateUrl()"></span>',
+      controller: ['$scope', function($scope) {
+        if (!$scope.value) {
+          $scope.value = [];
+        }
+        if (!$scope.prevIndices) {
+          $scope.prevIndices = [];
+        }
+        $scope.initArgs = $scope.getInitArgs();
+        $scope.choices = $scope.initArgs.choices;
+
+        if ($scope.selectedRank !== '' && $scope.selectedRank !== null) {
+          $scope.prevIndices.push(parseInt($scope.selectedRank));
```

Comment:
This logic seems incorrect. This function will be run only for the first time and then `prevIndices` will not be modified whenever the user selects some a rank. You need to keep it updated in `selectedItem()` method below. Also you don't need to keep track whole array of choices ever made. You can just keep track of max index yet.

Reply:
Done.